compiler-rt/lib/mpk_untrusted/alloc_site_handler.h

```diff
@@ -121,17 +123,37 @@ class AllocSite {
   // previous allocation sites for a given reallocated pointer.
   alloc_set_type& getAssociatedSet() { return associatedSet; }
 
+  // This struct is required for hashing the AllocSite properly in the unordered_set.
+  // Some Additional shuffling is used in the hashing of multiple values to ensure 
+  // that if the Function name and BasicBlock name happen to match, they do not 
+  // cancel each other out.
+  struct Hasher {
```

Comment:
Is this the same hashing logic from somewhere else? maybe the stl? if so, please site, if not is there anything special to note, aside from the comment above.

also, the details about shifting belong inside the implementation

the comment about being required is more or less fine, since you're just saying
```// required for std::unordered_set``` or equivalent

Reply:
Fixed via commit `fea9c81`.